src/state/slices/txHistorySlice/txHistorySlice.test.ts

```diff
@@ -36,33 +45,61 @@ describe('txHistorySlice', () => {
     it('should add new transactions', async () => {
       store.dispatch(txHistory.actions.clear())
 
+      const ethAccountSpecifier = `${EthSend.caip2}:0xdef1cafe`
+
       // new eth transaction (send)
-      store.dispatch(txHistory.actions.onMessage({ message: EthSend }))
+      store.dispatch(
+        txHistory.actions.onMessage({ message: EthSend, accountSpecifier: ethAccountSpecifier })
+      )
       expect(Object.values(store.getState().txHistory.ids).length).toBe(1)
 
       // duplicate eth transaction (send)
-      store.dispatch(txHistory.actions.onMessage({ message: EthSend }))
+      store.dispatch(
+        txHistory.actions.onMessage({ message: EthSend, accountSpecifier: ethAccountSpecifier })
+      )
       expect(Object.values(store.getState().txHistory.ids).length).toBe(1)
 
       // new eth transaction (receive)
-      store.dispatch(txHistory.actions.onMessage({ message: EthReceive }))
+      store.dispatch(
+        txHistory.actions.onMessage({ message: EthReceive, accountSpecifier: ethAccountSpecifier })
+      )
       expect(Object.values(store.getState().txHistory.ids).length).toBe(2)
 
       // eth data exists
       expect(store.getState().txHistory.byId[EthSend.txid]).toEqual(EthSend)
       expect(store.getState().txHistory.byId[EthReceive.txid]).toEqual(EthReceive)
 
+      const segwitNativeAccountSpecifier = `${BtcSend.caip2}:zpub`
+
       // new btc transaction (send)
-      store.dispatch(txHistory.actions.onMessage({ message: BtcSend }))
+      store.dispatch(
+        txHistory.actions.onMessage({
+          message: BtcSend,
+          accountSpecifier: segwitNativeAccountSpecifier
+        })
+      )
       expect(Object.values(store.getState().txHistory.ids).length).toBe(3)
 
       // duplicate btc transaction (send)
-      store.dispatch(txHistory.actions.onMessage({ message: BtcSend }))
+      store.dispatch(
+        txHistory.actions.onMessage({
+          message: BtcSend,
+          accountSpecifier: segwitNativeAccountSpecifier
+        })
+      )
       expect(Object.values(store.getState().txHistory.ids).length).toBe(3)
 
+      const segwitAccountSpecifier = `${BtcSend.caip2}:ypub`
+      // const segwitNativeAccountSpecifier = `${BtcSend.caip2}:zpub`
+
       // same btc transaction, different account type (send)
-      const BtcSendSegwit = { ...BtcSend, accountType: UtxoAccountType.SegwitNative }
-      store.dispatch(txHistory.actions.onMessage({ message: BtcSendSegwit }))
+      const BtcSendSegwit = { ...BtcSend, accountType: UtxoAccountType.SegwitP2sh }
```

Comment:
@DaoDev44 i'm pretty sure this test was wrong when i last updated it - the `txHistory.ids` length should remain the same, but we need to check that the tx id gets put against two different account Ids.

we should more generally add another test that tests tx ids against account ids to be sure it's behaving as expected.